src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Rules/PropertyPages/ApplicationPropertyPage.xaml

```diff
@@ -44,8 +56,23 @@
                   Persistence="ProjectFileWithInterception"
                   HasConfigurationCondition="False"/>
     </DynamicEnumProperty.DataSource>
+    <DynamicEnumProperty.Metadata>
+      <NameValuePair Name="VisibilityCondition" Value="(not (evaluated TargetMultipleFrameworks))" />
```

Comment:
Again, I think we need to specify the property's page.

```suggestion
      <NameValuePair Name="VisibilityCondition" Value="(not (evaluated Application TargetMultipleFrameworks))" />
```

Reply:
*If* we support cross-page visibility conditions I would propose we support both of the following:

`(evaluated PropertyName)` returns the evaluated value of the property PropertyName on the same page.
`(evaluated PageName PropertyName)` returns the evaluated value of the property PropertyName on the page PageName.

Reply:
To broaden this topic of discussion, what if we supported the following?

1. `Value="TargetMultipleFrameworks"`: produces the evaluated value of "TargetMultipleFrameworks" from the same page.
2. `Value="(evaluated TargetMultipleFrameworks)"`: produces the evaluated value of "TargetMultipleFrameworks" from the same page.
3. `Value="(unevaluated TargetMultipleFrameworks)"`: produces the unevaluated value of "TargetMultipleFrameworks" from the same page.
4. `Value="(evaluated OtherPage TargetMultipleFrameworks)"`: produces the evaluated value of "TargetMultipleFrameworks" from the "OtherPage" page.
5. `Value="(unevaluated OtherPage TargetMultipleFrameworks)"`: produces the unevaluated value of "TargetMultipleFrameworks" from the "OtherPage" page.

The idea here is to simplify the most common case (getting an evaluated value from a property on the same page), while still leaving open the option to get unevaluated values, or values from other pages.

Reply:
I believe we should support cross-page visibility conditions.

A motivating example of something I would like to add based on customer feedback is a bool property that forces a single "version" across the project (file, assembly, package, etc). This would likely be on the Application page, and could hide something on the Packaging page.

My preference is to require the page name in the expression. It makes the evaluation simpler, as there's no per-property context, so we can reuse the same environment across all properties when we update visibilities, avoiding a bunch of GC.

Reply:
Hmm, fair enough. OK, we can always add support for the simplified forms later. I'll update the PR.

Reply:
Your second comment wasn't visible to me when I wrote my reply.

I'm not sure we need to simplify the expressions here. Having a consistent form will help the occasional unfamiliar author stumble upon the correct pattern.

I'd vote for using the same syntax for qualified property names in both DependsOn and these expressions.